lib/presentation/bookPages/bloc/book_page_bloc.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:book/presentation/bookPages/bloc/book_page_events.dart';
+import 'package:book/presentation/bookPages/bloc/book_page_state.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class BookPagesBloc extends Bloc<BookPagesEvents, BookPagesState> {
+  BookPagesBloc() : super((InitialState())) {
+    on<NextPageEvent>(_onNextPage);
+    on<PreviousPageEvent>(_onPreviousPage);
+  }
+
+  Future<void> _onNextPage(
+      NextPageEvent event, Emitter<BookPagesState> emit) async {
+    emit(NextPage(currentIndex: event.currentIndex + 1));
```

Comment:
You don't have to forward current index through events and state for next and previous page, you can do the calculation in listener of BlocConsumer.